src/lib/Dungeon.js

```diff
@@ -282,137 +325,279 @@ class AutomationDungeon
+    }
 
-                let positionToCheck = { floor: playerCurrentPosition.floor, x: playerCurrentPosition.x, y: playerCurrentPosition.y - 1 };
-                let tileToCheck = DungeonRunner.map.board()[positionToCheck.floor][positionToCheck.y][positionToCheck.x];
+    /**
+     * @brief Marks boss and chest tiles in the passed cell
+     *
+     * @param cell: The cell to analyze, in the shape {x, y, floor, tile}
+     */
+    static __internal__markCell(cell)
+    {
+        const cellType = cell.tile.type();
+        if (cellType === GameConstants.DungeonTile.boss
+            || cellType === GameConstants.DungeonTile.ladder)
+        {
+            this.__internal__bossPosition = cell;
+        }
+        else if (cellType === GameConstants.DungeonTile.chest)
+        {
+            this.__internal__addChestPosition(cell);
+        }
+    }
 
-                // Don't consider not-visible tiles
-                if (!tileToCheck.isVisible)
-                {
-                    return;
-                }
+    /**
+     * @brief Marks relevant features in the current tile
+     */
+    static __internal__markCurrentTile()
+    {
+        const point = DungeonRunner.map.playerPosition();
+        this.__internal__markCell({ ...point, tile: DungeonRunner.map.board()[point.floor][point.y][point.x] });
+    }
 
-                // If the user has the flashlight, check for the boss or a chest presence on the tile above
-                if (tileToCheck.type() === GameConstants.DungeonTile.boss)
-                {
-                    this.__internal__bossPosition = positionToCheck;
-                    this.__internal__isCompleted = true;
-                }
-                else if (tileToCheck.type() === GameConstants.DungeonTile.chest)
-                {
-                    this.__internal__addChestPosition(positionToCheck);
-                }
-            }
+    /**
+     * @brief Marks relevant features in nearby tiles
+     *
+     * This should only be used if the player has flash unlocked, as it would cheat otherwise
+     */
+    static __internal__markNearbyTiles()
+    {
+        const point = DungeonRunner.map.playerPosition();
+        // Cant use the map.nearbyTiles() function as it doesn't return the coordinates
+        const nearbyCells = this.__internal__nearbyCells(point);
+        for (const cell of nearbyCells)
+        {
+            this.__internal__markCell(cell);
         }
-        // Else hide the menu and turn off the feature, if we're not in the dungeon anymore
-        else
+    }
+
+    static neighborsVectors = [
+        { x: 0, y: -1 },
+        { x: 0, y: 1 },
+        { x: -1, y: 0 },
+        { x: 1, y: 0 },
+    ];
+
+    /**
+     * @brief Returns the list of neighbouring cells
+     *
+     * This is similar to the map.nearbyTiles() function, but it returns the coordinates as well
+     * It cant be as short as the map.nearbyTiles() function we dont have the ?. operator in js to check for the bounds
```

Comment:
Are you sure about that ? 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

I tried in the firefox console and it worked as intended.